lib/otel/span-processor.js

```diff
@@ -73,8 +82,61 @@ module.exports = class NrSpanProcessor {
       this.reconcileDbAttributes({ segment, span })
     } else if (span.kind === SpanKind.PRODUCER) {
       this.reconcileProducerAttributes({ segment, span })
+    } else if (span.kind === SpanKind.CLIENT && (span.attributes[ATTR_HTTP_METHOD] || span.attributes[ATTR_HTTP_REQUEST_METHOD])) {
+      this.reconcileHttpExternalAttributes({ segment, span })
+    }
+  }
+
+  reconcileHttpExternalAttributes({ segment, span }) {
+    for (const [prop, value] of Object.entries(span.attributes)) {
+      let key = prop
+      let sanitized = value
+
+      if (key === ATTR_HTTP_REQUEST_METHOD || key === ATTR_HTTP_METHOD) {
+        key = 'procedure'
+      } else if (key === ATTR_SERVER_ADDRESS || key === ATTR_NET_PEER_NAME) {
+        key = 'hostname'
+        sanitized = this.sanitizeHostname(sanitized)
+        segment.addSpanAttribute(key, sanitized)
```

Comment:
these should only be added as span. this function is also then creating them as segment attributes